segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/ServiceStarter.java

```diff
@@ -166,18 +171,22 @@ private void attachStorage(ServiceBuilder builder) {
                 switch (storageChoice) {
                     case HDFS:
                         HDFSStorageConfig hdfsConfig = setup.getConfig(HDFSStorageConfig::builder);
-                        return new HDFSStorageFactory(hdfsConfig, setup.getExecutor());
+                        StorageMetricsBase metrics = new HDFSMetrics();
```

Comment:
If you do have a situation where you want to pass some external metrics, then my recommendation is to overload the constructor (one with and one without Metrics); the one without metrics just passes in the default implementation.